code/modules/capture_the_flag/ctf_game.dm

```diff
@@ -581,109 +422,105 @@
-		if (!team.ctf_enabled)
-			continue
-		scores += UNLINT("<span style='color: [team.team]'>[team.team] - [team.control_points]/[team.control_points_to_win]</span>\n")
+			to_chat(user, span_notice("Loading CTF..."))
 
-	balloon_alert_to_viewers(scores)
+			loading = CTF_LOADING_LOADING
+			if(activated_id == CTF_GHOST_CTF_GAME_ID) //Only ghost CTF supports map loading, if CTF is started by an admin elsewhere the map loader should not be used.
+				if(!GLOB.ctf_spawner.load_map(user))
+					to_chat(user, span_warning("CTF loading was cancelled"))
+					loading = CTF_LOADING_UNLOADED
+					return
+			loading = CTF_LOADING_LOADED
+		if (CTF_LOADING_LOADING)
+			to_chat(user, span_warning("CTF is loading!"))
 
-/obj/machinery/control_point/attackby(mob/user, params)
-	capture(user)
+			return
 
-/obj/machinery/control_point/attack_hand(mob/user, list/modifiers)
-	. = ..()
-	if(.)
-		return
-	capture(user)
+	var/ctf_enabled = FALSE
+	ctf_enabled = ctf_controller.toggle_ctf()
+	for(var/obj/machinery/power/emitter/emitter in ctf_area)
+		emitter.active = ctf_enabled
+	if(user)
+		message_admins("[key_name_admin(user)] has [ctf_enabled ? "enabled" : "disabled"] CTF!")
+	else if(automated)
+		message_admins("CTF has finished a round and automatically restarted.")
+		notify_ghosts("CTF has automatically restarted after a round finished in [initial(ctf_area.name)]!",'sound/effects/ghost2.ogg')
+	else
+		message_admins("The players have spoken! Voting has enabled CTF!")
+	if(!automated)
+		notify_ghosts("CTF has been [ctf_enabled? "enabled" : "disabled"] in [initial(ctf_area.name)]!",'sound/effects/ghost2.ogg')
 
-/obj/machinery/control_point/proc/capture(mob/user)
-	if(do_after(user, 30, target = src))
-		for(var/obj/machinery/capture_the_flag/team as anything in GLOB.ctf_panel.ctf_machines)
-			if(team.ctf_enabled && (user.ckey in team.team_members))
-				controlling = team
-				icon_state = "dominator-[team.team]"
-				for(var/mob/M in GLOB.player_list)
-					var/area/mob_area = get_area(M)
-					if(istype(mob_area, game_area))
-						to_chat(M, "<span class='userdanger [team.team_span]'>[user.real_name] has captured \the [src], claiming it for [team.team]! Go take it back!</span>")
-				break
+#undef CTF_LOADING_UNLOADED
+#undef CTF_LOADING_LOADING
+#undef CTF_LOADING_LOADED
 
+///Proc that identifies if something is a valid target for CTF related checks, checks if an object is a ctf barrier or has ctf component if they are a player.
 /proc/is_ctf_target(atom/target)
 	. = FALSE
 	if(istype(target, /obj/structure/barricade/security/ctf))
 		. = TRUE
 	if(ishuman(target))
-		var/mob/living/carbon/human/H = target
-		for(var/obj/machinery/capture_the_flag/CTF as anything in GLOB.ctf_panel.ctf_machines)
-			if(H in CTF.spawned_mobs)
-				. = TRUE
-				break
+		var/mob/living/carbon/human/human = target
+		if(human.mind?.GetComponent(/datum/component/ctf_player))
+			. = TRUE
```

Comment:
would rather you switch this into `return FALSE`/`return TRUE` instead of relying on funny dot operator

if you're spooked for runtimes and want to keep the dot operator, please include a `return .` at the very end of this proc.